tmt/steps/discover/__init__.py

```diff
@@ -463,16 +463,19 @@ def go(self, force: bool = False) -> None:
                 for test_origin in self.tests(enabled=True):
                     fmf_id = test_origin.test.fmf_id
 
-                    if not fmf_id.url:
-                        continue
+                    if not (fmf_id.url or self.opt('url')):
```

Comment:
```suggestion
                    if not (fmf_id.url or self.data.url):
```

Reply:
```suggestion
                    if not fmf_id.url:
```

I would keep the condition as this and state in the exception that the `--fmf-id` option requires a git repository. Let's keep things simple. I don't think we need to describe all the ways how to get it.